operators/pkg/instautoctrl/termination.go

```diff
@@ -179,7 +180,23 @@ func (r *InstanceTerminationReconciler) TerminateInstance(ctx context.Context, i
 	log := ctrl.LoggerFrom(ctx).WithName("termination")
 	log.Info("terminating instance")
 
-	instance.SetLabels(forge.InstanceAutomationLabelsOnTermination(instance.GetLabels()))
+	submissionRequired := false
+
+	template, err := RetrieveTemplate(ctx, r.Client, instance)
+
+	if err != nil {
+		log.Error(err, "failed retrieving template")
+		_, err := CheckEnvironmentValidity(instance, template)
```

Comment:
Suppose this should be within the if block below.

Reply:
Thanks, was getting mad on this